openpype/plugins/publish/extract_thumbnail.py

```diff
@@ -154,12 +209,42 @@ def _already_has_thumbnail(self, repres):
                 return True
         return False
 
+    def _get_explicit_repres_for_thumbnail(self, instance):
+        src_repres = instance.data.get("representations") or []
+        # This is mainly for Nuke where we have multiple representations for
+        #   one instance. We want to use only one representation for thumbnail
+        # first check if any of the representations have
+        # `need_thumbnail` in tags and add them to filtered_repres
+        need_thumb_repres = [
+            repre for repre in src_repres
+            if "need_thumbnail" in repre.get("tags", [])
+            if "publish_on_farm" not in repre.get("tags", [])
+        ]
+        if not need_thumb_repres:
+            return []
+
+        self.log.info(
```

Comment:
What will happen if you have more than one representation with `need_thumbnail`? From comments on this method one could think that it should actually return only one representation per instance, but the method name and return type suggest list?